### ISSUE TITLE
Need to check modified html files for 'use strict' directive

### ISSUE THREAD

**Issue by natduca** [catapult-project/catapult#1252]
The presubmit scripts used to check html files for 'use strict' directives. This was lost, probably by my cl to get rid of the code that tried to construct the full deps graph on presubmit.

A simpler variation of this would be, on presubmit, make the vulcanizer for the project, then just ask the vulcanizer to load the modified/added html files. If they are missing directives, they won't load.

**Comment by natduca** [comment]
this is probably p1 but we have a lot of p1 items. :) So, p1.5?

**Comment by anniesullie** [comment]
I'll leave this available while I'm hacking on the dev_server changes. If someone gets to this before I do, this seems like a good place to add a check: https://github.com/catapult-project/catapult/blob/master/build/js_checks.py

**Comment by natduca** [comment]
Yeah, not something you have to get to. If you look at js_checks.py, its a pile of glue code to talk to gjslint. I tend to think maybe that file should be gjslint_checks.py and we might want another file for html-parsing-based checks...

**Comment by anniesullie** [comment]
In the short term, should I add a test that just tries to vulcanize? Could also spit out the size to the perf dashboard when it is run on the main waterfall.

**Comment by anniesullie** [comment]
Ah, we already do have a test that vulcanizes (Trace2HTMLTests). But doesn't cover code outside of trace-viewer.

**Comment by natduca** [comment]
We used to have a presubmit that found all the things that looked like polymer modules, and said "vulcnize all these". I bet we could bring that back.

The key there is making sure to only find sourc files. THere are lots of html files that aren't modules that can't be treated as modules for vulcanizing: examples, third_party code, etc.

Probalby we can make a check up at the build level that asks every project for its GetSourceDirectories.
